src/store/reducers/usersReducer.ts

```diff
@@ -0,0 +1,31 @@
+import { IUser, Result } from '../interfaces';
+import { UsersAction } from '../types';
+import { UsersActionType } from '../action-creators/ActionTypes';
+
+const initialState: Result<Array<IUser>> = {
```

Comment:
I would recommend to move initialState setup into "constants.ts". Try to kee preducer as clean as possible.